### DISCUSSION TITLE
How to apply specific conversion rule on a field

### DISCUSSION THREAD

**Discussion by hantsy** [Kotlin/dataframe#289] [Q&A] [answered]
We received a csv from our partner and we need to parse the file. But we found there is an issue in this file, there are different formats for the date fields.

After reading the csv file, and use `convert` or `parse` to convert the field to a `LocalDate`.

```kotlin
DataFrame.readCSV(...)
  .convertTo<MyData>{
     parser{ LocalDate.parse(it, dateTimeFormatter)}
     convert{}
  }
  .toList
```  

The `MyData` is `@DataSchema` data class.

But here `parser` and `convert` are applied to all fields, how to bind to a specific field(defined by `@ColumnName`)?

**Comment by koperagen** [comment, marked as answer]
As something that 100% should work, I suggest 
```
DataFrame.readCSV(...).convert { "fieldName"<String>() }.with { LocalDate.parse(it, dateTimeFormatter) }.convertTo<MyData>()
```
